cmd/certsuite/upload/results_spreadsheet/results_spreadsheet.go

```diff
@@ -0,0 +1,367 @@
+}
+
+func readCSV(fp string) ([][]string, error) {
+	file, err := os.Open(fp)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	reader := csv.NewReader(file)
+	records, err := reader.ReadAll()
+	if err != nil {
+		return nil, err
+	}
+	return records, nil
+}
+
+func createSheetsAndDriveServices() (sheetService *sheets.Service, driveService *drive.Service, err error) {
+	ctx := context.Background()
+	b, err := os.ReadFile("cmd/certsuite/upload/results_spreadsheet/credentials.json")
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to read client secret file: %v", err)
+	}
+
+	// If modifying these scopes, delete your previously saved token.json.
+	config, err := google.ConfigFromJSON(b, sheets.SpreadsheetsScope, drive.DriveScope)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to parse client secret file to config: %v", err)
+	}
+	client, err := getClient(config)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to get client: %v", err)
+	}
+
+	sheetSrv, err := sheets.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to retrieve Sheets client: %v", err)
+	}
+
+	driveSrv, err := drive.NewService(ctx, option.WithHTTPClient(client))
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to retrieve Drive client: %v", err)
+	}
+
+	return sheetSrv, driveSrv, nil
+}
+
+func prepareRecordsForSpreadSheet(records [][]string) []*sheets.RowData {
+	var rows []*sheets.RowData
+	for _, row := range records {
+		var rowData []*sheets.CellData
+		for _, col := range row {
+			var val string
+			// cell content cannot exceed 50,000 letters.
+			if len(col) > cellContantLimit {
+				col = col[:cellContantLimit]
+			}
+			// use space for empty values to avoid cells overlapping
+			if col == "" {
+				val = " "
+			}
+			// avoid line breaks in cell
+			val = strings.ReplaceAll(strings.ReplaceAll(col, "\r\n", " "), "\n", " ")
+
+			rowData = append(rowData, &sheets.CellData{
+				UserEnteredValue: &sheets.ExtendedValue{StringValue: &val},
+			})
+		}
+		rows = append(rows, &sheets.RowData{Values: rowData})
+	}
+	return rows
+}
+
+func createSingleWorkloadRawResultsSheet(rawResultsSheet *sheets.Sheet, workloadName string) (*sheets.Sheet, error) {
+	// Initialize sheet with headers
+	filteredRows := []*sheets.RowData{{Values: []*sheets.CellData{
+		{UserEnteredValue: &sheets.ExtendedValue{StringValue: stringToPointer("Owner/TechLead Conclusion")}},
+		{UserEnteredValue: &sheets.ExtendedValue{StringValue: stringToPointer("Next Step Actions")}},
+	}}}
+	filteredRows[0].Values = append(filteredRows[0].Values, rawResultsSheet.Data[0].RowData[0].Values...)
+
+	headers := getHeadersFromSheet(rawResultsSheet)
+	indices, err := getHeaderIndicesByColumnNames(headers, []string{"CNFName"})
+	if err != nil {
+		return nil, err
+	}
+	workloadIndex := indices[0]
+
+	// add to sheet only rows of given workload name
+	for _, row := range rawResultsSheet.Data[0].RowData[1:] {
```

Comment:
Is `len(rawResultsSheet.Data[0].RowData) > 1` checked anywhere before calling this func? Otherwise it's a good pre-condition to add at the very begining of this one.